### PR TITLE
Add computed variable Router.routes

### PR DESCRIPTION
**Pull request by adam-fowler**
Returns a list of all the endpoints added to the router

### CODE REVIEW

File: Sources/Hummingbird/Router/Router.swift [hunk 1]
```diff
@@ -128,3 +128,25 @@ public struct RouterOptions: OptionSet, Sendable {
...
+
+extension Router {
+    /// Route description
+    public struct RouteDescription: CustomStringConvertible {
+        /// Route path
+        public let path: RouterPath
+        /// Route method
+        public let method: HTTPRequest.Method
+
+        public var description: String { "\(method): \(path)" }
```
Comment (Joannis) [inline Sources/Hummingbird/Router/Router.swift:140]: Can we print it without colon? E.G.

```
GET /users/
POST /users/
```

```suggestion
        public var description: String { "\(method) \(path)" }
```

File: Sources/Hummingbird/Router/TrieRouter.swift [hunk 2]
```diff
@@ -91,3 +91,20 @@ import HummingbirdCore
         }
     }
 }
+
+extension RouterPathTrieBuilder.Node {
+    /// Return list of paths and associated values in trie node
+    /// - Parameter prefix: Prefix for path
+    /// - Returns: Array of path values pairs
+    func values(prefix: RouterPath = "/") -> [(value: Value, path: RouterPath)] {
```
Comment (Joannis) [inline Sources/Hummingbird/Router/TrieRouter.swift:99]: I don't think we care about performance here, but for the sake of having mentioned it we should consider a syntax with `into values: inout [(value: Value, path: RouterPath)]`. This allows us to limit (re)allocs of arrays.

**Review by Joannis — APPROVED** [review]

**Comment by Joannis** [comment]
Two nits

File: Sources/Hummingbird/Router/TrieRouter.swift (hunk 2 above)
Comment (adam-fowler) [reply Sources/Hummingbird/Router/TrieRouter.swift:99]: For the sake of readability and given we don't care about performance here I'll keep it as is

**Pull request merged**